### ISSUE TITLE
Block clicking page elements

### ISSUE THREAD

**Issue by 9o** [nkbt/react-page-click#1]
Hey Nik,

Thanks so much for the great React component!

In the demo, when you click somewhere in the page to close the modal while also clicking the "open popup" button, it immediately reopens the window. I wish it would block clicking page elements by creating an overlay, but I realise that might not be in the scope of this component. What do you think? Maybe an example with using an overlay should be in the docs?

**Comment by nkbt** [comment]
Thank you, @9o 

This component itself is meant to be the leanest possible solution that could be re-used in different scenarios. 

That is a good idea to add overlay into an example since it is one of the common use-cases.
Another one is - just a simple tooltip-popover.

I'll kindly appreciate PRs on other examples as well! And will update docs soon.

**Comment by nkbt** [comment]
@9o, I refactored example, so now it is a separate Modal example #2

**Comment by rpominov** [comment]
The best workaround I found for this is to add `onMouseDown` with `e.preventDefault()` handler to elements we want to ignore. It's not perfect anyway, but works for me.

```js
const toggle = () => {
  this.setState(s => ({opened: !s.opened}))
}

<div>
  <button onMouseDown={e => {e.preventDefault}} onClick={toggle}>toggle</button>
  {this.state.opened && <PageClick onClick={toggle}>...</PageClick>}
</div>
```

**Comment by nkbt** [comment]
@rpominov thanks, that should work (when you don't have a "shade" over the screen). I think I am using something similar in my projects. There is not much to be done inside PageClick itself though, since it should not really override behaviour of outside components.

**Comment by rpominov** [comment]
> There is not much to be done inside PageClick itself though, since it should not really override behaviour of outside components.

Yeah, I also don't see how it could be solved inside PageClick.

**Comment by nkbt** [comment]
![Good news everone!](http://i.giphy.com/3o7abA4a0QCXtSxGN2.gif)

This was accidentally fixed by #14
